### PR TITLE
frontend updates (still WIP) and graph queries

### PR DESCRIPTION
**Pull request by gabsong**
Frontend is not complete, but we got the GraphQL queries and would like to merge this so @6rivu can work on the Mint page, and I can work on the Profile page separately.

### CODE REVIEW

File: src/components/Dapp.jsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export default function Dapp() {
   async function handleFollow() {
```
Comment (gabsong) [inline src/components/Dapp.jsx]: @6rivu there's a UX bug, which is that after sending the follow transaction, we want to clear the text inputs. I'll let you figure that out.

File: contracts/ProfileToken.sol [hunk 2]
```diff
@@ -15,18 +15,27 @@ contract ProfileToken is
...
+
     event NewProfile(uint256 tokenId, address owner);
 
     constructor() {
     }
 
+    //check if connected wallet-address already has a tokenID
+    function checkEntry(address _address) external pure returns(bool) {
+        return tokenExists[_address];
+    }
```
Comment (gabsong) [inline contracts/ProfileToken.sol]: I think we can call the ERC721 balanceOf(address) instead. If the return value is greater than 0, then you know the address has ownership of 1 token or more.

File: contracts/ProfileToken.sol [hunk 3]
```diff
@@ -15,18 +15,27 @@ contract ProfileToken is
 
     Counters.Counter public _tokenIds;
 
+    mapping (address => bool) public tokenExists;
```
Comment (gabsong) [inline contracts/ProfileToken.sol]: not needed if using ERC721.balanceOf

**Review by ewerx — APPROVED** [review]

**Review by 6rivu — APPROVED** [review]

**Comment by brutoshi** [comment]
OK

**Pull request merged**